vscode/qodana/src/core/client/activities.ts

```diff
@@ -1,66 +1,142 @@
-import { LanguageClient } from "vscode-languageclient/node";
+import { LanguageClient, State } from "vscode-languageclient/node";
 import * as vscode from "vscode";
 
 import { Auth } from "../auth";
-import { getReportFile } from "../report";
-import { CLOUD_REPORT_LOADED, NEW_REPORT_AVAILABLE, NO, SHOW_PROBLEMS, YES } from "../messages";
-import config from "../config";
+import { openReportById, openReportByProjectId } from "../report";
+import { CLOUD_REPORT_LOADED, SHOW_PROBLEMS } from "../messages";
+import config, {
+    CONF_PROJ_ID, IS_DEBUG,
+    WS_BASELINE_ISSUES,
+    WS_COMPUTED_PREFIX,
+    WS_OPENED_REPORT,
+    WS_REPORT_ID
+} from "../config";
 import sarif from "../sarif";
 import telemetry from "../telemetry";
-import { extensionInstance } from "../extension";
+import { Events, ReportFileEvent, UrlCallbackEvent } from "../events";
+import { BaselineToggle } from "../menuitems/BaselineToggle";
+import { QodanaState } from "../menuitems/QodanaState";
 
-async function announceWorkspaceFolder(client: LanguageClient, context: vscode.ExtensionContext) {
+export function onUrlCallback(context: vscode.ExtensionContext, auth: Auth) {
+    Events.instance.onUrlCallback(async (event: UrlCallbackEvent) => {
+        let authorized = auth.getAuthorized();
+        if (authorized) {
+            await openReportById(event.projectId, event.reportId, context, authorized);
+        } else {
+            telemetry.errorReceived('#onUrlCallback exception');
```

Comment:
I think that here we may still at least navigate to the location if found. WDYT?

Reply:
I think we already opening it in any case, maybe without opened report